wordpress-mobile/WordPress-iOS
# Remove PostSignUpInterstitialViewController

Address the feedback from https://github.com/wordpress-mobile/WordPress-iOS/pull/23439#discussion_r1696533167.

## Changes

- Remove `PostSignUpInterstitialViewController`
- Rework `NoSitesView` to use the new `EmptyStateView`

## To test:

In both WordPress and Jetpack apps:

- Login using an existing account with no sites
- Create a new account
- Login using an account with a single site and delete it

Verify that the updated `NoSitesView` is shown when you complete the login. 

## Notes

The current production behavior is to show `PostSignUpInterstitialViewController` (screenshot on the left) when you complete a login with no sites or sign up. If you tap "Skip", you land on the dashboard and see the `NoSitesView` view, which serves the same function as the previous screen.

<img width="320" alt="Screenshot 2024-07-30 at 10 35 38 AM" src="https://github.com/user-attachments/assets/c1ec64b7-2ea8-466c-9368-0543a79403c9">

<img width="320" alt="Screenshot 2024-07-30 at 10 38 17 AM" src="https://github.com/user-attachments/assets/44dd95cd-6e9a-432a-8257-86bd82822860">

`PostSignUpInterstitialViewController` has a couple of issues:

- As Tony [noted](https://github.com/wordpress-mobile/WordPress-iOS/pull/23439#discussion_r1696533167), button taps are communicated to the app using notifications, which is not something you want to see in an app
- It uses old graphics and design of components not found anywhere else in the app
- It doesn't specify that the site is created on WordPress.com

The agreed upon solution was to remove the first screen `PostSignUpInterstitialViewController` to simplify the flow and make the app simpler and easier to maintain.

When you login with no sites or create a new account, you will now immediately lang on dashboard and see an nice updated empty state view that uses `EmptyStateView` added in https://github.com/wordpress-mobile/WordPress-iOS/pull/23443. It now also uses the new `AddSiteAlertViewModel` added in https://github.com/wordpress-mobile/WordPress-iOS/pull/23454 to avoid duplicating business logic and localizable strings.

<img width="320" alt="Screenshot 2024-08-01 at 9 38 55 AM" src="https://github.com/user-attachments/assets/788cb37e-2216-4935-93fd-b06c29788ac4">

<img width="320" alt="Screenshot 2024-08-01 at 9 35 52 AM" src="https://github.com/user-attachments/assets/f31e3d7d-31c4-4052-890d-00e356a0930d">



## Regression Notes
1. Potential unintended areas of impact


2. What I did to test those areas of impact (or what existing automated tests I relied on)


3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:


Testing checklist:
